### ISSUE TITLE
In segments, rules are sometimes duplicated.

### ISSUE THREAD

**Issue by novakzaballa** [SolidStateGroup/bullet-train-api#4388]
### How are you running Flagsmith

- [ ] Self Hosted with Docker
- [ ] Self Hosted with Kubernetes
- [ ] SaaS at flagsmith.com
- [X] Some other way (add details in description below)

### Describe the bug

In the segment modal, the rules can be duplicated, when create 100 rules

### Steps To Reproduce

1. Go to segments
2. Select any segment
3. Create 89 rules, and update the segment
4. Refresh the page
5. Create the 11 rules more and update the segment
6. In the network you can see more than 100 rules, and some duplicated

### Expected behavior

A user should only create 100 rules within a segment, and these should not be cloned

### Screenshots

![image](https://github.com/user-attachments/assets/b7431244-9f3d-4930-bf83-af35507e5819)

**Comment by kyle-ssg** [comment]
@rolodato  I think I remember you seeing this a while ago, I'm not sure if this issue can be closed now ?

**Comment by rolodato** [comment]
Yes, I believe this was fixed in https://github.com/Flagsmith/flagsmith/issues/4725
